internal/common/resource.go

```diff
@@ -303,6 +314,14 @@ func (r *reconcileBuilder) createOrUpdateWithImmutableSpec(obj client.Object, f
 	}
 
 	existing := obj.DeepCopyObject().(client.Object)
+
+	// During an upgrade to v0.19.0 we might encounter virt-operator attempting
+	// to reconcile the same set of common-instancetype resources. If configured
+	// ignore these requests to reconcile such objects once owned by virt-operator
+	if r.options.IgnoreObjectOwnedByVirtOperator && isOwnedByVirtOperator(existing) {
```

Comment:
Put this at the start of `Reconcile` instead to short-circuit the whole reconciliation?

Reply:
Moving this earlier would cause another client lookup for every object for every reconcile. We can move this but I'm not sure it's worth it.

In terms of the `VersionCache` yeah we can avoid this but AFAICT there's no impact of this being registered?

Reply:
Alright, then let's keep it here.

Just wanted to make sure we don't create any issues by caching resources we do not own.

Reply:
> Moving this earlier would cause another client lookup for every object for every reconcile. We can move this but I'm not sure it's worth it.

Client lookups are cheap, because everything is cached in memory. The API client watches the cluster, and any `Get()` or `List()` function calls just read the cahce.